docs/PRIORITIES.md

```diff
@@ -1,78 +1,52 @@
+# High priority
+
 ## Fees
+
 - Collection
-  - Gas price based on parameter
-  - (which gets changed automatically)
-  - https://github.com/cosmos/cosmos-sdk/issues/1921
-  - Per block gas usage as %
-  - Windowing function
-    - Block N,
-    - For Block N-x ~ N, get average of %
-  - Should take into account time.
-  - Standard for querying this price // needs to be used by UX.
+  - Simple flat fee set in-config by validators & full nodes - ref [#1921](https://github.com/cosmos/cosmos-sdk/issues/1921)
 - Distribution
-  - MVP: 1 week, 1 week for testing.
+  - "Piggy bank" fee distribution - ref [#1944](https://github.com/cosmos/cosmos-sdk/pull/1944) (spec)
+- Reserve pool
+  - Collects in a special field for now, no spending
 
 ## Governance v2
-- V1 is just text proposals
-  - Software upgrade stuff
-    - https://github.com/cosmos/cosmos-sdk/issues/1734#issuecomment-407254938
-    - https://github.com/cosmos/cosmos-sdk/issues/1079
-- We need to test auto-flipping w/ threshold voting power.
-- Super simple:
-  - Only use text proposals
-  - On-chain mechanism for agreeing on when to "flip" to new functionality
+
+- Simple software upgrade proposals
+  - Implementation described in [#1079](https://github.com/cosmos/cosmos-sdk/issues/1079)
+  - Agree upon a block height to switch to new version
```

Comment:
Block height seems in contradiction with implementation described in https://github.com/cosmos/cosmos-sdk/issues/1079 ?

Also block height introduces a risk of halting when switch happens.

Reply:
It does introduce a risk of halting, but halting isn't catastrophic - we just will need to wait for more of the network to upgrade.

cc @sunnya97 for context on the implementation

Reply:
Is the solution described in https://github.com/cosmos/cosmos-sdk/issues/1079 that more difficult to implement compared to block height. It's just the mechanism by which we know when to switch, it does not change the implementation of the switch itself, or does it?

Halting may not be *catastrophic*, but it's pretty bad for the image of the network ("Headline: Cosmos tried to update, and is now stuck..."). Especially after token transfer is enabled.

Reply:
It's more difficult to implement threshold signaling than block height; yes, the latter just requires that the new software version switch at a given block height while the former requires that we track the percent of the network signaling.

I'm not sure what you mean by "image of the network" - I think it's a prudent and intentional design choice to prioritize safety over liveness. If the network agrees upon an upgrade, but half really upgrades and half does not, halting seems like the safe thing to do so we can figure out what's going on! As soon as part of either half switches to the other version, the network will resume.

In any case this was agreed upon by @jaekwon in an SDK meeting (unfortunately not recorded, sorry) so I think we'll need another discussion if we want to change it again.

Reply:
Hmm I don't follow the safety vs liveness argument here. Both solution favour safety over liveness.  I understand it's more difficult, I was just arguing about the level of added complexity vs the benefits we can get from threshold mechanism. 

In any case no problem if this was agreed in meeting! Maybe something to reconsider soon after MVP is implemented :)

Reply:
Definitely agreed on considering this post-launch.

Reply:
I think there might have been some miscommunication or I just forgot... let me write a simple spec up and then we can discuss that.  Will try to get this in my Tuesday 12pm.

Reply:
Well, I'm late for it.  But I'll put it on my plate for tonight.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/2116